besu/src/main/java/org/hyperledger/besu/cli/options/unstable/PkiBlockCreationOptions.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.hyperledger.besu.cli.options.unstable;
+
+import static java.util.Arrays.asList;
+import static org.hyperledger.besu.cli.DefaultCommandValues.MANDATORY_FILE_FORMAT_HELP;
+
+import org.hyperledger.besu.cli.util.CommandLineUtils;
+import org.hyperledger.besu.ethereum.api.tls.FileBasedPasswordProvider;
+import org.hyperledger.besu.pki.config.PkiKeyStoreConfiguration;
+
+import java.nio.file.Path;
+import java.util.Optional;
+
+import org.apache.logging.log4j.Logger;
+import picocli.CommandLine;
+import picocli.CommandLine.Option;
+import picocli.CommandLine.ParameterException;
+
+public class PkiBlockCreationOptions {
+
+  @Option(
+      names = {"--Xpki-block-creation-enabled"},
+      hidden = true,
+      description = "Enable PKI integration (default: ${DEFAULT-VALUE})")
+  Boolean enabled = false;
+
+  @Option(
+      names = {"--Xpki-block-creation-keystore-type"},
+      hidden = true,
+      paramLabel = "<NAME>",
+      description = "PKI service keystore type. Required if PKI Block Creation is enabled.")
+  @SuppressWarnings({"FieldCanBeFinal", "FieldMayBeFinal"})
+  String keyStoreType = PkiKeyStoreConfiguration.DEFAULT_KEYSTORE_TYPE;
+
+  @Option(
+      names = {"--Xpki-block-creation-keystore-file"},
+      hidden = true,
+      paramLabel = MANDATORY_FILE_FORMAT_HELP,
+      description = "Keystore containing key/certificate for PKI PKI Block Creation.")
```

Comment:
It could be both. When we read the certificate, we use `keystore.getCertificateChain`. If it has intermediate certificates in the keystore, we will retrieve them and add to the CMS.